tutorials/time-series-modeling/Lab_05.md

```diff
@@ -0,0 +1,392 @@
+
+In simpler terms, we are breaking up more complex curves into a set of simplified cosine waves with an additive term. Mark Jakubauskas has an informative [paper](https://www.isprs.org/proceedings/xxxiii/congress/part4/384_xxxiii-part4.pdf) that breaks down the process, and there are many resources that elaborate more on the math behind harmonic models. Building a harmonic model is used in remote sensing applications because of its flexibility in accounting for cyclicality with simple, reproducible shapes. If there is a seasonal trend in the data, the ordered nature of a cosine curve can likely approximate it.  
+
+ To fit this model to the time series, set $\omega$=1 (one cycle per unit time) and use ordinary least squares regression as the metric of error reduction. 
+
+First, add the harmonic variables (the third and fourth terms of equation 2) to the image collection.
+```javascript
+// Use these independent variables in the harmonic regression.
+var harmonicIndependents = ee.List(['constant', 't', 'cos', 'sin']);
+// Add harmonic terms as new image bands.
+var harmonicLandsat = filteredLandsat.map(function(image) {
+  var timeRadians = image.select('t').multiply(2 * Math.PI);
+    return image
+      .addBands(timeRadians.cos().rename('cos'))
+      .addBands(timeRadians.sin().rename('sin'));
+  });
+```
+
+Fit the model with a linear trend, using the `linearRegression()` reducer:
+
+```javascript
+var harmonicTrend = harmonicLandsat
+  .select(harmonicIndependents.add(dependent))
+  // The output of this reducer is a 4x1 array image.
+  .reduce(ee.Reducer.linearRegression({
+  numX: harmonicIndependents.length(), 
+  numY: 1
+  }));
+```
+
+Plug the coefficients into equation 2 in order to get a time series of fitted values:
+
+```javascript
+// Turn the array image into a multi-band image of coefficients.
+var harmonicTrendCoefficients = harmonicTrend.select('coefficients') 
+		.arrayProject([0]) 				
+		.arrayFlatten([harmonicIndependents]);
+// Compute fitted values.
+var fittedHarmonic = harmonicLandsat.map(function(image) { 
+  return image.addBands(  
+  image.select(harmonicIndependents)   
+  .multiply(harmonicTrendCoefficients)   
+  .reduce('sum')   
+  .rename('fitted'));
+});
+// Plot the fitted model and the original data at the ROI.
+print(ui.Chart.image.series(fittedHarmonic.select(['fitted','NDVI']), roi,
+                  ee.Reducer.mean(), 30)  
+  .setSeriesNames(['NDVI', 'fitted'])  
+  .setOptions({   
+  title: 'Harmonic Model: Original and Fitted Values',   
+  lineWidth: 1,   
+  pointSize: 3,}));
+```
+
+The harmonic model (red data points) does an adequate job of predicting the data - There is a datapoint in Feb 2021 that is significantly lower, but this appears to be an outlier. Additionally, the model misses a significant dip in July 2015, although this might be due to irregular climate conditions - other years did not have the same dip.  
+
+> **Question 1**: Upload the resulting graphic and interpret it.
```

Comment:
Remove the questions to be consistent with the other tutorials

Reply:
Question removed